### PR TITLE
Show eMMC lifetime

### PR DESCRIPTION
**Pull request by ludeeus**
## Proposed change

Adds eMMC lifetime stats to the host card.
Backend: https://github.com/home-assistant/supervisor/pull/2413

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:
image

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by agners** [comment]
Currently we only get this information from the eMMC. But the idea would be to also get the information from other types of SSDs such as S-ATA or NVMe attached SSDs. S-ATA (S.M.A.R.T.) and the NVMe specific interface to get SSD lifetime show a percentage with digit accuracy. eMMC is really the outlier here, by expressing a range :(

For the supervisor interface I now just went with the pessimistic value (e.g. 10 for 0-10%). Just showing 10% is maybe a bit annoying as a new device will show 10% when started the first time, not a nice user experience :(

I guess we can show it eMMC specific for now on the frontend and change once we have support for other SSDs.

**Comment by ludeeus** [comment]
I tried the non-range first, but as you mention showing 10% on a new device is not good.
I'd say we keep it as is for now, and change when we have other cases.

**Review by agners — APPROVED** [review]
Not a TS expert, but null seems to be properly handled and range calculation seems correct!

File: hassio/src/system/hassio-host-info.ts [hunk 1]
```diff
@@ -150,6 +150,17 @@ class HassioHostInfo extends LitElement {
               : ""}
           </div>
           <div>
+            ${this.supervisor.host.disk_life_time
```
Comment (bramkragten) [inline hassio/src/system/hassio-host-info.ts:153]: If it is `0` we don't show it now, is that intended?

File: hassio/src/system/hassio-host-info.ts [hunk 2]
```diff
@@ -150,6 +150,17 @@ class HassioHostInfo extends LitElement {
...
+              ? html` <ha-settings-row>
+                  <span slot="heading">
+                    eMMC Lifetime Used
+                  </span>
+                  <span slot="description">
+                    ${this.supervisor.host.disk_life_time - 10}% -
```
Comment (bramkragten) [inline hassio/src/system/hassio-host-info.ts:159]: I guess it is not 0, as this would be -10 :-)

**Comment by bramkragten** [comment]
If this is emmc specific handling, should we have the value on a different prop, or indicate with a prop it is an emmc value? `disk_life_time` -> `emmc_disk_life_time`

File: hassio/src/system/hassio-host-info.ts (hunk 1 above)
Comment (ludeeus) [reply hassio/src/system/hassio-host-info.ts:153]: the lowest value returned is 10

**Comment by ludeeus** [comment]
When support for more disk types are added, a new API key (`disk_type`) will be added/needed as well, we can then use that to do the correct calculations where needed.

File: hassio/src/system/hassio-host-info.ts (hunk 1 above)
Comment (balloob) [reply hassio/src/system/hassio-host-info.ts:153]: that's an assumption :D I would check against undefined.

File: hassio/src/system/hassio-host-info.ts (hunk 2 above)
Comment (balloob) [reply hassio/src/system/hassio-host-info.ts:159]: lol such an Apple move.

**Comment by pvizeli** [comment]
The OS call for this is a bit hacky. I think we can expose that better under `/hardware/disk` with a nice disk overview and can be later the base for using USB disk or so.

File: hassio/src/system/hassio-host-info.ts (hunk 1 above)
Comment (ludeeus) [reply hassio/src/system/hassio-host-info.ts:153]: Undefined is not used, but i get your point

**Review by bramkragten — APPROVED** [review]

**Pull request merged**